datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -344,8 +344,8 @@ impl PhysicalExpr for BinaryExpr {
         let right_interval = children[1];
 
         let (left, right) = if self.op.is_logic_operator() {
-            // TODO: Currently, this implementation only supports the AND operator
-            //       and does not require any further propagation. In the future,
+            // TODO: Currently, this implementation only supports the AND/OR operator
```

Comment:
I would appreciate any suggestions if any changes are needed here in propagate_constraints as well

Reply:
I'm slightly changing the API here. It will be more understandable and easy to add new supports. Since the current version is written by overfitting the AND operator, adding OR support may be a challenge. Undoubtedly, it will be easier to add it after my refactoring PR arrives.

Reply:
Sounds good -- I don't fully understand what propagate constraints is used for. I think postponing any extra work until your refactoring is complete sounds like a good idea to me